### DISCUSSION TITLE
%sys (system) CPU usage discussion

### DISCUSSION THREAD

**Discussion by dison4linux** [blakeblackshear/frigate#5780] [Ask A Question]
I've got Frigate dialed in fairly well (running 0.12-beta10 at the moment) CPU wise, at least %user CPU usage.
What I'd like to discuss is %sys CPU usage.
I don't know the Linux side of things as well as I'd like, but what I've come to understand is that %sys CPU is more or less incidental CPU usage that goes along with the actual CPU usage that you want to accomplish something, being the %user.
Even back when I was running version 0.11.1 the %system usage was higher than I would like.
On a given day my %user averages around 30%... maybe at little less at night when not much is going on, maybe a little more on a super windy day, but I feel like 30% is pretty good.  BTW, I have an i5-9400.
Logging CPU usage over time, I have an almost perfectly flat line of about 20% system CPU usage, if I temporarily stop the frigate container that goes to almost zero.
Various frigate processes each use a minimal amount of %sys CPU usage, but it all adds up.  The detect process seems to be the only one that uses noteworthy %sys CPU usage on its own.

```
04:59:26 PM   UID       PID    %usr %system  %guest   %wait    %CPU   CPU  Command
04:59:26 PM     0   1085366    0.00    0.00    0.00    0.00    0.00     3  frigate.logger
04:59:26 PM     0   1085374    0.45    0.22    0.00    0.02    0.67     2  frigate.detecto
04:59:26 PM     0   1085376    0.03    0.02    0.00    0.02    0.05     3  frigate.output
04:59:26 PM     0   1085379    0.00    0.00    0.00    0.00    0.00     4  frigate.process
04:59:26 PM     0   1085381    0.09    0.03    0.00    0.01    0.12     0  frigate.process
04:59:26 PM     0   1085382    0.21    0.06    0.00    0.03    0.27     1  frigate.process
04:59:26 PM     0   1085384    0.04    0.01    0.00    0.01    0.05     3  frigate.process
04:59:26 PM     0   1085385    0.01    0.00    0.00    0.00    0.02     3  frigate.process
04:59:26 PM     0   1085387    0.01    0.00    0.00    0.00    0.02     5  frigate.process
04:59:26 PM     0   1085389    0.04    0.02    0.00    0.00    0.06     0  frigate.process
04:59:26 PM     0   1085390    0.01    0.00    0.00    0.00    0.02     2  frigate.process
04:59:26 PM     0   1085392    0.02    0.00    0.00    0.01    0.02     5  frigate.process
04:59:26 PM     0   1085393    0.05    0.02    0.00    0.01    0.07     4  frigate.process
04:59:26 PM     0   1085395    0.00    0.01    0.00    0.00    0.01     3  frigate.capture
04:59:26 PM     0   1085398    0.02    0.01    0.00    0.00    0.03     5  frigate.capture
04:59:26 PM     0   1085403    0.04    0.06    0.00    0.00    0.10     5  frigate.capture
04:59:26 PM     0   1085406    0.01    0.01    0.00    0.00    0.02     0  frigate.capture
04:59:26 PM     0   1085413    0.01    0.01    0.00    0.00    0.02     1  frigate.capture
04:59:26 PM     0   1085419    0.01    0.01    0.00    0.00    0.02     5  frigate.capture
04:59:26 PM     0   1085422    0.01    0.01    0.00    0.00    0.02     1  frigate.capture
04:59:26 PM     0   1085430    0.01    0.01    0.00    0.00    0.02     0  frigate.capture
04:59:26 PM     0   1085433    0.01    0.01    0.00    0.00    0.02     1  frigate.capture
04:59:26 PM     0   1085440    0.02    0.02    0.00    0.00    0.04     5  frigate.capture
```

**Comment by NickM-27** [comment]
It's not really clear what the question is here? I'd just stick with the general CPU % which is % of a single core. Frigate's system page in the webUI shows this for each process and makes it easy to compare configurations.

**Comment by dison4linux** [comment]
I guess the question is... is it normal and/or to be expected that the %sys is so high when running Frigate?  It is unusual to me.
I've got several other apps running in docker on other systems and only %user shoots up when its "busy" while %sys stays consistently low.
Whereas when running Frigate, I get a constant almost 20% %sys.
If there's a Linux admin out there that can say "oh yeah, that means you don't have enough RAM" or "I've seen this before, it was high IRQs from the Network card, because its a crappy RealTek NIC" ... then I'd appreciate that insight...
If it is what it is.... it is what it is.

**Reply by NickM-27** [reply]
https://stackoverflow.com/a/556411 should hopefully give you some more info on this.

System usage makes sense given that ffmpeg will be using the GPU to run decoding and moving frames between CPU and GPU memory among other things
